commit 83e42a0035718914efad2fe04080fd6c579e9ef2
Author: Michael Paquier <michael@paquier.xyz>
Date:   Sat Oct 1 15:28:02 2022 +0900

    doc: Fix some grammar and typos
    
    This fixes some areas related to logical replication and custom RMGRs.
    
    Author: Ekaterina Kiryanova
    Discussion: https://postgr.es/m/fa4773f1-1396-384a-bcd7-85b5e013f399@postgrespro.ru
    Backpatch-through: 15

diff --git a/doc/src/sgml/custom-rmgr.sgml b/doc/src/sgml/custom-rmgr.sgml
index acf5077d751..2893016cef3 100644
--- a/doc/src/sgml/custom-rmgr.sgml
+++ b/doc/src/sgml/custom-rmgr.sgml
@@ -64,22 +64,22 @@ typedef struct RmgrData
 <programlisting>
 /*
  * Register a new custom WAL resource manager.
  *
  * Resource manager IDs must be globally unique across all extensions. Refer
- * to https://wiki.postgresql.org/wiki/CustomWALResourceManager to reserve a
+ * to https://wiki.postgresql.org/wiki/CustomWALResourceManagers to reserve a
  * unique RmgrId for your extension, to avoid conflicts with other extension
  * developers. During development, use RM_EXPERIMENTAL_ID to avoid needlessly
  * reserving a new ID.
  */
 extern void RegisterCustomRmgr(RmgrId rmid, RmgrData *rmgr);
 </programlisting>
   <function>RegisterCustomRmgr</function> must be called from the
   extension module's <link linkend="xfunc-c-dynload">_PG_init</link> function.
   While developing a new extension, use <literal>RM_EXPERIMENTAL_ID</literal>
-  for <parameter>rmid</parameter>. When you ready to release the extension to
-  users, reserve a new resource manager ID at the <ulink
+  for <parameter>rmid</parameter>. When you are ready to release the extension
+  to users, reserve a new resource manager ID at the <ulink
   url="https://wiki.postgresql.org/wiki/CustomWALResourceManagers">Custom WAL
   Resource Manager</ulink> page.
  </para>
 
  <para>
diff --git a/doc/src/sgml/func.sgml b/doc/src/sgml/func.sgml
index e82077292c6..b2bdbc7d1cc 100644
--- a/doc/src/sgml/func.sgml
+++ b/doc/src/sgml/func.sgml
@@ -25552,11 +25552,11 @@ SELECT collation for ('foo' COLLATE "de_DE");
         specified process ID.  This function can send the request to
         backends and auxiliary processes except logger.  These memory contexts
         will be logged at
         <literal>LOG</literal> message level. They will appear in
         the server log based on the log configuration set
-        (See <xref linkend="runtime-config-logging"/> for more information),
+        (see <xref linkend="runtime-config-logging"/> for more information),
         but will not be sent to the client regardless of
         <xref linkend="guc-client-min-messages"/>.
        </para></entry>
       </row>
 
diff --git a/doc/src/sgml/logical-replication.sgml b/doc/src/sgml/logical-replication.sgml
index 77be4c37e7f..e98538e5400 100644
--- a/doc/src/sgml/logical-replication.sgml
+++ b/doc/src/sgml/logical-replication.sgml
@@ -686,22 +686,22 @@ test_sub=# SELECT * FROM t3;
     <emphasis>any</emphasis> of the expressions will be replicated. This means all
     the other row filters for the same table become redundant if:
     <itemizedlist>
      <listitem>
       <para>
-       one of the publications has no row filter.
+       One of the publications has no row filter.
       </para>
      </listitem>
      <listitem>
       <para>
-       one of the publications was created using <literal>FOR ALL TABLES</literal>.
+       One of the publications was created using <literal>FOR ALL TABLES</literal>.
        This clause does not allow row filters.
       </para>
      </listitem>
      <listitem>
       <para>
-       one of the publications was created using
+       One of the publications was created using
        <literal>FOR TABLES IN SCHEMA</literal> and the table belongs to
        the referred schema. This clause does not allow row filters.
       </para>
      </listitem>
     </itemizedlist></para>
diff --git a/src/backend/access/transam/rmgr.c b/src/backend/access/transam/rmgr.c
index 8ed69244e39..3b6de3aa04e 100644
--- a/src/backend/access/transam/rmgr.c
+++ b/src/backend/access/transam/rmgr.c
@@ -87,11 +87,11 @@ RmgrNotFound(RmgrId rmid)
 
 /*
  * Register a new custom WAL resource manager.
  *
  * Resource manager IDs must be globally unique across all extensions. Refer
- * to https://wiki.postgresql.org/wiki/CustomWALResourceManager to reserve a
+ * to https://wiki.postgresql.org/wiki/CustomWALResourceManagers to reserve a
  * unique RmgrId for your extension, to avoid conflicts with other extension
  * developers. During development, use RM_EXPERIMENTAL_ID to avoid needlessly
  * reserving a new ID.
  */
 void
